lib/measures/tbd/resources/psi.rb

```diff
@@ -61,79 +64,256 @@ def append(k)
+      h[:sill]            = @set[p].has_key?(:sill)
+      h[:sillconcave]     = @set[p].has_key?(:sillconcave)
+      h[:sillconvex]      = @set[p].has_key?(:sillconvex)
+      h[:jamb]            = @set[p].has_key?(:jamb)
+      h[:jambconcave]     = @set[p].has_key?(:jambconcave)
+      h[:jambconvex]      = @set[p].has_key?(:jambconvex)
+      h[:corner]          = @set[p].has_key?(:corner)
+      h[:cornerconcave]   = @set[p].has_key?(:cornerconcave)
+      h[:cornerconvex]    = @set[p].has_key?(:cornerconvex)
+      h[:parapet]         = @set[p].has_key?(:parapet)
+      h[:partyconcave]    = @set[p].has_key?(:parapetconcave)
+      h[:parapetconvex]   = @set[p].has_key?(:parapetconvex)
+      h[:party]           = @set[p].has_key?(:party)
+      h[:partyconcave]    = @set[p].has_key?(:partyconcave)
+      h[:partyconvex]     = @set[p].has_key?(:partyconvex)
+      h[:grade]           = @set[p].has_key?(:grade)
+      h[:gradeconcave]    = @set[p].has_key?(:gradeconcave)
+      h[:gradeconvex]     = @set[p].has_key?(:gradeconvex)
+      h[:balcony]         = @set[p].has_key?(:balcony)
+      h[:balconyconcave]  = @set[p].has_key?(:balconyconcave)
+      h[:balconyconvex]   = @set[p].has_key?(:balconyconvex)
+      h[:rimjoist]        = @set[p].has_key?(:rimjoist)
+      h[:rimjoistconcave] = @set[p].has_key?(:rimjoistconcave)
+      h[:rimjoistconvex]  = @set[p].has_key?(:rimjoistconvex)
+      @has[p] = h
+
+      v = {}
+      v[:joint]    = 0; v[:transition]      = 0; v[:fenestration]   = 0
+      v[:head]     = 0; v[:headconcave]     = 0; v[:headconvex]     = 0
+      v[:sill]     = 0; v[:sillconcave]     = 0; v[:sillconvex]     = 0
+      v[:jamb]     = 0; v[:jambconcave]     = 0; v[:jambconvex]     = 0
+      v[:corner]   = 0; v[:cornerconcave]   = 0; v[:cornerconvex]   = 0
+      v[:parapet]  = 0; v[:parapetconcave]  = 0; v[:parapetconvex]  = 0
+      v[:party]    = 0; v[:partyconcave]    = 0; v[:partyconvex]    = 0
+      v[:grade]    = 0; v[:gradeconcave]    = 0; v[:gradeconvex]    = 0
+      v[:balcony]  = 0; v[:balconyconcave]  = 0; v[:balconyconvex]  = 0
+      v[:rimjoist] = 0; v[:rimjoistconcave] = 0; v[:rimjoistconvex] = 0
+
+      v[:joint]           = @set[p][:joint]           if h[:joint]
+      v[:transition]      = @set[p][:transition]      if h[:transition]
+      v[:fenestration]    = @set[p][:fenestration]    if h[:fenestration]
+      v[:head]            = @set[p][:fenestration]    if h[:fenestration]
+      v[:headconcave]     = @set[p][:fenestration]    if h[:fenestration]
+      v[:headconvex]      = @set[p][:fenestration]    if h[:fenestration]
+      v[:sill]            = @set[p][:fenestration]    if h[:fenestration]
+      v[:sillconcave]     = @set[p][:fenestration]    if h[:fenestration]
+      v[:sillconvex]      = @set[p][:fenestration]    if h[:fenestration]
+      v[:jamb]            = @set[p][:fenestration]    if h[:fenestration]
+      v[:jambconcave]     = @set[p][:fenestration]    if h[:fenestration]
```

Comment:
Ok, I think I see how this works.  First jambconcave gets fenestration, then jamb (if it exists), then jambconcave (if that exists)

Reply:
You got it.